tests/hypervolume_tests/test_box_decomposition.py

```diff
@@ -0,0 +1,140 @@
+    """See Section 4.2 of https://arxiv.org/pdf/1510.01963"""
+    points = _generate_uniform_samples(n_trials, n_objectives, seed)
+    loss_values = points / np.maximum(_EPS, np.sum(points**2, axis=-1))[:, np.newaxis]
+    return _extract_pareto_sols(loss_values)
+
+
+def _generate_convex_instances(n_trials: int, n_objectives: int, seed: int) -> np.ndarray:
+    """See Section 4.2 of https://arxiv.org/pdf/1510.01963"""
+    points = _generate_uniform_samples(n_trials, n_objectives, seed)
+    loss_values = 1 - points / np.maximum(_EPS, np.sum(points**2, axis=-1))[:, np.newaxis]
+    return _extract_pareto_sols(loss_values)
+
+
+def _generate_linear_instances(n_trials: int, n_objectives: int, seed: int) -> np.ndarray:
+    """See Section 4.2 of https://arxiv.org/pdf/1510.01963"""
+    points = _generate_uniform_samples(n_trials, n_objectives, seed)
+    loss_values = points / np.maximum(_EPS, np.sum(points, axis=-1))[:, np.newaxis]
+    return _extract_pareto_sols(loss_values)
+
+
+def _calculate_hypervolume_improvement(
+    lbs: np.ndarray, ubs: np.ndarray, new_points: np.ndarray
+) -> np.ndarray:
+    diff = np.maximum(0.0, ubs - np.maximum(new_points[..., np.newaxis, :], lbs))
+    # The minimization version of Eq. (1) in https://arxiv.org/pdf/2006.05078.
+    return np.sum(np.prod(diff, axis=-1), axis=-1)
+
+
+def _verify_exact_hypervolume_improvement(pareto_sols: np.ndarray) -> None:
+    ref_point = np.max(pareto_sols, axis=0)
+    ref_point = np.maximum(ref_point * 1.1, ref_point * 0.9)
+    hv = compute_hypervolume(pareto_sols, ref_point, assume_pareto=True)
+    for i in range(pareto_sols.shape[0]):
+        # LOO means leave one out.
+        loo = np.arange(pareto_sols.shape[0]) != i
+        correct = hv - compute_hypervolume(pareto_sols[loo], ref_point)
+        lbs, ubs = get_non_dominated_box_bounds(pareto_sols[loo], ref_point)
+        out = _calculate_hypervolume_improvement(lbs, ubs, pareto_sols[np.newaxis, i])
+        assert out.shape == (1,)
+        assert np.isclose(out[0], correct)
+
+
+@pytest.mark.parametrize(
+    "gen",
+    (
+        _generate_concave_instances,
+        _generate_convex_instances,
+        _generate_instances_with_negative,
+        _generate_linear_instances,
+    ),
+)
+@pytest.mark.parametrize("n_objectives", [2, 3, 4])
+def test_exact_box_decomposition(gen: InstanceGenerator, n_objectives: int) -> None:
+    n_trials = 30 if n_objectives == 4 else 60
+    pareto_sols = gen(n_objectives=n_objectives, n_trials=n_trials, seed=42)
+    _verify_exact_hypervolume_improvement(pareto_sols)
+
+
+@pytest.mark.parametrize(
+    "gen",
+    (
+        _generate_concave_instances,
+        _generate_convex_instances,
+        _generate_instances_with_negative,
+        _generate_linear_instances,
+    ),
+)
+@pytest.mark.parametrize("n_objectives", [2, 3, 4])
+def test_box_decomposition_general_position(gen: InstanceGenerator, n_objectives: int) -> None:
```

Comment:
Since this unit test corresponds to the relaxation of the ``general position'' assumption (sec. 2.4), the naming, the naming is inverted.
```suggestion
def test_box_decomposition_non_general_position(gen: InstanceGenerator, n_objectives: int) -> None:
```